crates/iceberg/src/io/file_io.rs

```diff
@@ -95,9 +95,14 @@ impl FileIO {
     /// # Arguments
     ///
     /// * path: It should be *absolute* path starting with scheme string used to construct [`FileIO`].
-    pub async fn remove_all(&self, path: impl AsRef<str>) -> Result<()> {
+    pub async fn remove_dir_all(&self, path: impl AsRef<str>) -> Result<()> {
```

Comment:
[this comment](https://github.com/apache/iceberg-rust/issues/1118#issuecomment-2766198351) also mentions adding a `remove_dir` function for removing empty dir, do we also want to add that?